src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -378,23 +371,21 @@ void AndroidDeviceControllerWrapper::AsyncSetKeyValue(const char * key, const ch
 
     err = N2J_NewStringUTF(GetJavaEnv(), key, keyString);
     SuccessOrExit(err);
-    err = N2J_NewStringUTF(GetJavaEnv(), value, valueString);
+
+    err = N2J_NewStringUTF(GetJavaEnv(), static_cast<const char *>(value), size, valueString);
```

Comment:
Is this correct?  Does this handle embedded null (which is not really valid UTF-8), byte sequences that look like unpaired surrogates, etc, etc correctly?

I just tested like so in jshell, and I think this does not work right:
```
jshell> byte[] bytes = { -19, -96, -127 }
bytes ==> byte[3] { -19, -96, -127 }

jshell> new String(bytes, "UTF-8").getBytes()
$6 ==> byte[3] { -17, -65, -67 }
```
That's handing in an unpaired-surrogate-as-fake-UTF-8 byte sequence and what comes out is U+FFFD REPLACEMENT CHARACTER, because that thing is not valid UTF-8.

Reply:
I only did compilation check on this. Expectation is to eventually use KVS. Will look into having base64 encoding for now.

Reply:
Java is now moved to use KVS